DirectEmployers/MyJobs
# Improved monitoring of things that were recently identified as in need of improvement.

## What's this PR do?
* Adds a request queuing stat to the traces (so we can tell how long an individual trace was queueing) and as a metric (so we can see how all requests are doing).
* Gets rid of New Relic middleware and moves what New Relic middleware was monitoring to the monitoring file.
* Adds a decorator to saved search sending tasks that tracks the args/kwargs passed into the saved search task for debugging purposes. What is tracked looks something like this:
`{'contains_pss': False, 'args': (<@task: tasks.send_search_digest of default:0x7f129eb85790 (v2 compatible)>, 57L), 'frequency': u'D', 'is_digest': False, 'user_email': u'alice@example.com'}`
* Ensures that site id is attached to every trace.

## Where should the reviewer start?
* Primary changes are in monitoring.py
* Are all references to NewRelic middleware removed?
* Are there any other tasks that need monitored.

## How should this be manually tested?
For saved searches, I just added a print statement to show what log_dict was sending and then ran tests.
For request queueing of it relies on Apache, so you can't, you just have to take my word for it that it should work.
For the rest, turn on Datadog locally and make a few requests.

## Any background context you want to provide?
New Relic is gone on Friday, so I figured why not remove the middleware now?
Request queueing additions are a push toward feature parity with New Relic.
Saved search changes are a response to lack of debugging information encountered last night.


## PR Checklist
It doesn't look like any are relevant.
